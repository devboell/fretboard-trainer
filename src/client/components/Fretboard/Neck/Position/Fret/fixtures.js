import { merge } from 'ramda'

const noteE2 = 'E2'
const noteA2 = 'A2'
const noteC4 = 'C4'
const locE2 = { str: 5, pos: 0 }
const locA2 = { str: 4, pos: 0 }
const selectedE2 = { note: noteE2, status: 'selected', label: 'root' }
const selectedC4 = { note: noteC4, status: 'selected', label: 'root' }
const selectedNotes = [selectedE2, selectedC4]
const selectedLocations = []

const testCases = isSelected => [
  {
    description: isSelected
      ? 'no content, selection highlight'
      : 'no content, no highlight',
    context: {
      showNotes: false,
      showOctaves: false,
      showSelection: false,
      showEnharmonics: false,
    },
  },
  {
    description: isSelected
      ? 'note content, selection highlight'
      : 'note content, no highlight',
    context: {
      showNotes: true,
      showOctaves: false,
      showSelection: false,
      showEnharmonics: false,
    },
  },
  {
    description: isSelected
      ? 'note content, selection highlight'
      : 'note content, octave highlight',
    context: {
      showNotes: true,
      showOctaves: true,
      showSelection: false,
      showEnharmonics: false,
    },
  },
  {
    description: isSelected
      ? 'selection content, selection highlight'
      : 'note content, octave highlight',
    context: {
      showNotes: true,
      showOctaves: true,
      showSelection: true,
      showEnharmonics: false,
    },
  },
]

const defaultContext = {
  clickAction: () => {},
}

export default [
  {
    description: 'E2, selected note, skinType=boxes',
    props: { note: noteE2, loc: locE2 },
    context: merge(defaultContext, {
      skinType: 'boxes',
      selectedNotes,
      selectedLocations,
    }),
    testCases: testCases(true),
  },
  {
    description: 'E2, selected note, skinType=strings',
    props: { note: noteE2, loc: locE2 },
    context: merge(defaultContext, {
      skinType: 'strings',
      selectedNotes,
      selectedLocations,
    }),
    testCases: testCases(true),
  },
  {
    description: 'A2, selected note, skinType=boxes',
    props: { note: noteA2, loc: locA2 },
    context: merge(defaultContext, {
      skinType: 'boxes',
      selectedNotes,
      selectedLocations,
    }),
    testCases: testCases(false),
  },
  {
    description: 'A2, unselected note, skinType=strings',
    props: { note: noteA2, loc: locA2 },
    context: merge(defaultContext, {
      skinType: 'strings',
      selectedNotes,
      selectedLocations,
    }),
    testCases: testCases(false),
  },
]

