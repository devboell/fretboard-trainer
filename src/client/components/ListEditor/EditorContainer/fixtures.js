import { createStore, combineReducers } from 'redux'

import reducer, { initialState as initialEditorState } from './reducer'
import { CREATE_QUIZ, UPDATE_QUIZ, DELETE_QUIZ } from './mutations'

export const quiz3 = {
  id: '3',
  name: 'quiz three',
  type: 'pc',
  __typename: 'Quiz',
}

export const updatedQuiz = {
  id: '3',
  name: 'updated',
  type: 'pc',
  __typename: 'Quiz',
}

export const createdQuiz = {
  id: '7',
  name: 'new',
  type: 'pc',
  __typename: 'Quiz',
}

export const crudMocks = [
  {
    request: {
      query: CREATE_QUIZ,
      variables: {
        name: 'new', type: 'pc', __typename: 'Quiz',
      },
    },
    result: { data: { createQuiz: createdQuiz } },
  },
  {
    request: {
      query: UPDATE_QUIZ,
      variables: {
        id: '3', name: 'updated', type: 'pc',
      },
    },
    result: { data: { updateQuiz: updatedQuiz } },
  },
  {
    request: {
      query: DELETE_QUIZ,
      variables: { id: '3' },
    },
    result: { data: { deleteQuiz: '3' } },
  },
]

const initialState = {
  editor: initialEditorState,
}

export const store = createStore(combineReducers({ editor: reducer }), initialState)
