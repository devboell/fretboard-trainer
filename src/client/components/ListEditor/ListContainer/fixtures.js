import { createStore, combineReducers } from 'redux'

import reducer, { initialState as initialListState } from './reducer'
import QUIZZES from './queries'

export const quizzes = [
  {
    id: '1',
    name: 'quiz one',
    type: 'pc',
    __typename: 'Quiz',
  },
  {
    id: '2',
    name: 'quiz two',
    type: 'pc',
    __typename: 'Quiz',
  },
  {
    id: '3',
    name: 'quiz three',
    type: 'pc',
    __typename: 'Quiz',
  },
  {
    id: '4',
    name: 'quiz four',
    type: 'pc',
    __typename: 'Quiz',
  },
  {
    id: '5',
    name: 'quiz five',
    type: 'pc',
    __typename: 'Quiz',
  },
  {
    id: '6',
    name: 'quiz six',
    type: 'pc',
    __typename: 'Quiz',
  },
]

export const quiz3 = {
  id: '3',
  name: 'quiz three',
  type: 'pc',
  __typename: 'Quiz',
}

export const updatedQuiz = {
  id: '3',
  name: 'updated',
  type: 'pc',
  __typename: 'Quiz',
}

export const createdQuiz = {
  id: '7',
  name: 'new',
  type: 'pc',
  __typename: 'Quiz',
}

export const quizzesMocks = [
  {
    request: { query: QUIZZES },
    result: { data: { quizzes } },
  },
]

const initialState = {
  list: initialListState,
}

export const store = createStore(combineReducers({ list: reducer }), initialState)
