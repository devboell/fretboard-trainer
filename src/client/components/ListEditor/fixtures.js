import { createStore, combineReducers } from 'redux'

import listReducer, { initialState as initialListState } from './ListContainer/reducer'
import editorReducer, { initialState as initialEditorState } from './EditorContainer/reducer'
import QUIZZES from './ListContainer/queries'
import {
  CREATE_QUIZ,
  UPDATE_QUIZ,
  DELETE_QUIZ,
} from './EditorContainer/mutations'

const tuning = 'standard'
const width = 13

export const quizzes = [
  {
    id: '1',
    name: 'quiz one',
    type: 'pc',
    tuning,
    width,
    __typename: 'Quiz',
  },
  {
    id: '2',
    name: 'quiz two',
    type: 'pc',
    tuning,
    width,
    __typename: 'Quiz',
  },
  {
    id: '3',
    name: 'quiz three',
    type: 'pc',
    tuning,
    width,
    __typename: 'Quiz',
  },
  {
    id: '4',
    name: 'quiz four',
    type: 'pc',
    tuning,
    width,
    __typename: 'Quiz',
  },
  {
    id: '5',
    name: 'quiz five',
    type: 'pc',
    tuning,
    width,
    __typename: 'Quiz',
  },
  {
    id: '6',
    name: 'quiz six',
    type: 'pc',
    tuning,
    width,
    __typename: 'Quiz',
  },
]

export const quiz3 = {
  id: '3',
  name: 'quiz three',
  type: 'pc',
  tuning,
  width,
  __typename: 'Quiz',
}

export const newQuiz = {
  name: 'new',
  type: 'pc',
  tuning,
  width,
  __typename: 'Quiz',
}

export const updatedQuiz = {
  id: '3',
  name: 'updated',
  type: 'pc',
  tuning,
  width,
}

export const createdQuiz = {
  id: '7',
  name: 'new',
  type: 'pc',
  tuning,
  width,
  __typename: 'Quiz',
}


export const quizzesMocks = [
  {
    request: { query: QUIZZES },
    result: { data: { quizzes } },
  },
]

export const crudMocks = [
  {
    request: {
      query: CREATE_QUIZ,
      variables: newQuiz,
    },
    result: { data: { createQuiz: createdQuiz } },
  },
  {
    request: {
      query: UPDATE_QUIZ,
      variables: updatedQuiz,
    },
    result: { data: { updateQuiz: updatedQuiz } },
  },
  {
    request: {
      query: DELETE_QUIZ,
      variables: { id: '3' },
    },
    result: { data: { deleteQuiz: '3' } },
  },
]

const initialState = {
  list: initialListState,
  editor: initialEditorState,
}

export const store = createStore(combineReducers({
  list: listReducer,
  editor: editorReducer,
}), initialState)
