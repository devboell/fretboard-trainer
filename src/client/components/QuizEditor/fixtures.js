import { createStore, combineReducers } from 'redux'

import reducer, { initialState as initialEditorState } from './reducer'
import QUIZZES from './queries'

export const quizzes = [
  {
    id: '1',
    name: 'quiz one',
    type: 'pc',
    __typename: 'Quiz',
  },
  {
    id: '2',
    name: 'quiz two',
    type: 'pc',
    __typename: 'Quiz',
  },
  {
    id: '3',
    name: 'quiz three',
    type: 'pc',
    __typename: 'Quiz',
  },
  {
    id: '4',
    name: 'quiz four',
    type: 'pc',
    __typename: 'Quiz',
  },
  {
    id: '5',
    name: 'quiz five',
    type: 'pc',
    __typename: 'Quiz',
  },
  {
    id: '6',
    name: 'quiz six',
    type: 'pc',
    __typename: 'Quiz',
  },
]

export const quiz3 = {
  id: '3',
  name: 'quiz three',
  type: 'pc',
  __typename: 'Quiz',
}

export const quizzesMocks = [
  {
    request: { query: QUIZZES },
    result: { data: { quizzes } },
  },
]

const initialState = {
  editor: initialEditorState,
}

export const store = createStore(combineReducers({ editor: reducer }), initialState)
