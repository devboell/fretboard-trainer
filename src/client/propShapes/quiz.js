import pt from 'prop-types'


export const panelModeShape = pt.shape({
  id: pt.string,
  question: pt.string,
  answer: pt.string,
})

export const quizShape = pt.shape({
  id: pt.string,
  name: pt.string,
  type: pt.string,
  tuning: pt.string,
  width: pt.number,
  panelModes: pt.arrayOf(panelModeShape),
})
