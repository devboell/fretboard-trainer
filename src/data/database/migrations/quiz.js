export function up(knex, Promise) {
  return Promise.all([
    knex.schema.createTable('Quiz', (table) => {
      table.increments('id').notNullable()
      table.string('name').notNullable()
      table.string('description').notNullable()
      table.enum('type', ['pc', 'pitch', 'interval']).notNullable()
      table.enum('tuning', ['standard']).notNullable()
      table.integer('width').notNullable()
      table.integer('allAnswers').notNullable()
      table.integer('allowIncorrect').notNullable()
      table.integer('useTimer').notNullable()
      table.integer('time').notNullable()
      table.integer('showNotes').notNullable()
      table.integer('highlightOctaves').notNullable()
    }),
  ])
}

export function down(knex, Promise) {
  return Promise.all([
    knex.schema.dropTable('Quiz'),
  ])
}
