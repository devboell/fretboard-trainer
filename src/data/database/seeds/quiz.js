const quizzes = `
  INSERT INTO 'Quiz' VALUES
  (
    1,
    'Notes I',
    'Very easy training to help memorize the notes on the fretboard. The fretboard displays all the note names, so they are easy to find. This is just a warm-up.',
    'pc',
    'standard',
    13,
    0,
    1,
    0,
    0,
    1,
    0
  ),
  (
    2,
    'Notes II',
    'no description',
    'pc',
    'standard',
    13,
    1,
    1,
    1,
    5,
    0,
    0
  ),
  (
    3,
    'quiz three',
    'no description',
    'pitch',
    'standard',
    13,
    1,
    1,
    1,
    5,
    0,
    1
  ),
  (
    4,
    'quiz four',
    'no description',
    'pitch',
    'standard',
    13,
    1,
    1,
    1,
    5,
    0,
    0
  ),
  (
    5,
    'quiz five',
    'no description',
    'pitch',
    'standard',
    13,
    1,
    1,
    1,
    5,
    0,
    0
  ),
  (
    6,
    'quiz six',
    'no description',
    'pitch',
    'standard',
    13,
    1,
    1,
    1,
    5,
    0,
    0
  )
`

export function seed(knex) { // eslint-disable-line import/prefer-default-export
  return knex('Quiz').truncate()
    .then(() =>
      knex.raw(quizzes))
}
