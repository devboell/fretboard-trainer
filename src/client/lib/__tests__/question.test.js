import { take } from 'ramda'
import { quizValues1 as pcQuiz } from 'fixtures/quiz'
import getQuestion from '../question'

const mockTake = take

jest.mock('lodash/fp', () => ({
  shuffle: jest.fn(arr => arr),
  sample: jest.fn(arr => mockTake(1, arr)[0]),
  sampleSize: jest.fn(size => arr => mockTake(size, arr)),
}))

it('pcQuestion', () => {
  expect(getQuestion(pcQuiz)).toMatchSnapshot()
})

it('invalid question', () => {
  const invalidQuiz = { ...pcQuiz, type: 'invalid' }
  expect(getQuestion(invalidQuiz)).toBeNull()
})
