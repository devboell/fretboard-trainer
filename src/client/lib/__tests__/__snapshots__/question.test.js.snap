// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pc question 1`] = `
Object {
  "evaluation": Object {
    "entity": Object {
      "name": "E",
      "notes": null,
    },
    "locs": Array [
      Object {
        "pos": 0,
        "str": 0,
      },
      Object {
        "pos": 0,
        "str": 5,
      },
      Object {
        "pos": 2,
        "str": 3,
      },
      Object {
        "pos": 5,
        "str": 1,
      },
      Object {
        "pos": 7,
        "str": 4,
      },
      Object {
        "pos": 9,
        "str": 2,
      },
      Object {
        "pos": 12,
        "str": 0,
      },
      Object {
        "pos": 12,
        "str": 5,
      },
    ],
  },
  "panels": Object {
    "answer": Object {
      "choices": Array [
        Object {
          "name": "B",
          "notes": null,
        },
        Object {
          "name": "G",
          "notes": null,
        },
        Object {
          "name": "D",
          "notes": null,
        },
        Object {
          "name": "E",
          "notes": null,
        },
      ],
      "locs": Array [],
    },
    "question": Object {
      "entity": Object {
        "name": "E",
        "notes": null,
      },
      "locs": Array [
        Object {
          "pos": 0,
          "str": 0,
        },
      ],
    },
  },
}
`;

exports[`pitch question 1`] = `
Object {
  "evaluation": Object {
    "entity": Object {
      "name": "E",
      "notes": null,
    },
    "locs": Array [
      Object {
        "pos": 0,
        "str": 0,
      },
      Object {
        "pos": 0,
        "str": 5,
      },
      Object {
        "pos": 2,
        "str": 3,
      },
      Object {
        "pos": 5,
        "str": 1,
      },
      Object {
        "pos": 7,
        "str": 4,
      },
      Object {
        "pos": 9,
        "str": 2,
      },
      Object {
        "pos": 12,
        "str": 0,
      },
      Object {
        "pos": 12,
        "str": 5,
      },
    ],
  },
  "panels": Object {
    "answer": Object {
      "choices": Array [
        Object {
          "name": "B",
          "notes": null,
        },
        Object {
          "name": "G",
          "notes": null,
        },
        Object {
          "name": "D",
          "notes": null,
        },
        Object {
          "name": "E",
          "notes": null,
        },
      ],
      "locs": Array [],
    },
    "question": Object {
      "entity": Object {
        "name": "E",
        "notes": null,
      },
      "locs": Array [
        Object {
          "pos": 0,
          "str": 0,
        },
      ],
    },
  },
}
`;
